tcmur_device.c

```diff
@@ -86,17 +87,28 @@ int __tcmu_reopen_dev(struct tcmu_device *dev, bool in_lock_thread)
 	pthread_mutex_lock(&rdev->state_lock);
 	rdev->flags &= ~TCMUR_DEV_FLAG_IS_OPEN;
 	ret = -EIO;
+	tcmu_dev_flags_set(dev, TCMU_DEV_FLAG_REOPENING);
 	while (ret != 0 && !(rdev->flags & TCMUR_DEV_FLAG_STOPPING)) {
 		pthread_mutex_unlock(&rdev->state_lock);
 
+		/* Avoid busy loop ? */
+		sleep(1);
```

Comment:
Let's move the sleep to below the open call and and do it if ret != 0.

Reply:
Yes, will update it.